src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2623,6 +2623,18 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
 
             var oldState = this.State;
             this.State = state;
+            var oldVariableSlot = _variableSlot;
+            _variableSlot = PooledDictionary<VariableIdentifier, int>.GetInstance();
```

Comment:
>_variableSlot [](start = 12, length = 13)

Should we also capture and reset ```variableBySlot``` and ```nextVariableSlot```?

Reply:
Good catch, thanks.

---
In reply to: [535415461](https://github.com/dotnet/roslyn/pull/49765#discussion_r535415461) [](ancestors = 535415461)